src/unknown_0255E0.c

```diff
@@ -911,25 +1010,52 @@ LevelModel *func_8002C7C4(void) {
     return gCurrentLevelModel;
 }
 
-GLOBAL_ASM("asm/non_matchings/unknown_0255E0/func_8002C7D4.s")
+void func_8002C7D4(void) {
+    s32 i;
+
+    func_8000B290();
+    if (D_8011D384 != 0) {
+        func_800B7D20();
+    }
+    for(i = 0; i < gCurrentLevelModel->numberOfTextures; i++) {
+        free_texture(gCurrentLevelModel->textures[i].texture);
+    }
+    free_from_memory_pool(D_8011D30C);
+    free_from_memory_pool(D_8011D370);
+    free_from_memory_pool(D_8011D374);
+    free_sprite((Sprite* ) gCurrentLevelModel->unk20);
+    for(i = 0; i < 4; i++) {
```

Comment:
I usually think about making any loops like this use MAXCONTROLLERS instead of 4, as I'm reasonably certain this is about the number of players. Could make it easier in some future where someone wants to redefine the number of players for some reason. Up to you though.